src/service/action-impl.js

```diff
@@ -200,6 +204,17 @@ export class ActionService {
         this.addChangeDetails_(event);
         this.trigger(dev().assertElement(event.target), name, event);
       });
+    } else if (name == 'input-debounced') {
+      const inputDebounced = debounce(this.ampdoc.win, event => {
+        const target = dev().assertElement(event.target);
+        this.trigger(target, name, /** @type {!Event} */ (event));
```

Comment:
We're gonna hit bugs later on with this typecast. Why not create a DeferredEvent type?